### PR TITLE
Create Block: Update templates to use APIs introduced in WP 6.1

### PR DESCRIPTION
**Pull request by gziolo**
## What?

Now that we have support for `render` in `block.json` coming to WordPress 6.1, we can integrate that into `@wordpress/create-block` to improve the developer experience.

**This PR will have to wait to land until we get closer to the WordPress 6.1 release date (November 1st).**

## How?

Add a new `render` field to `defaultValues` in external templates to let folks define the path to the PHP file that's going to be used with dynamic blocks. I also updated all dynamic variants to use his newly introduced config option.

## Testing Instructions

The following should scaffold a dynamic block that works without the build step.

```bash
npx wp-create-block --variant dynamic --template es5 dynamic-example
```

This is how you can test the version with the build step for the tutorial in Block Editor Handbook:

```bash
npx wp-create-block --no-wp-scripts --variant dynamic --template @wordpress/create-block-tutorial-template dynamic-example-tutorial
cd dynamic-example-tutorial
../node_modules/.bin/wp-scripts build
```

### CODE REVIEW

File: packages/create-block/lib/templates/es5/$slug.php.mustache [hunk 1]
```diff
@@ -70,32 +70,12 @@ function {{namespaceSnakeCase}}_{{slugSnakeCase}}_block_init() {
 	);
 
 	register_block_type(
-		'{{namespace}}/{{slug}}',
+		$dir,
 		array(
 			'editor_script' => '{{namespace}}-{{slug}}-block-editor',
 			'editor_style'  => '{{namespace}}-{{slug}}-block-editor',
 			'style'         => '{{namespace}}-{{slug}}-block',
```
Comment (gziolo) [inline packages/create-block/lib/templates/es5/$slug.php.mustache:77]: For now, I had to leave it as is, but I hope we can figure out a way to move all JS and CSS registration to `block.json` when someone opts out from using build tools. Tracked in https://github.com/WordPress/gutenberg/issues/40447.

File: packages/create-block/lib/templates/es5/index.js.mustache [hunk 2]
```diff
@@ -33,53 +33,6 @@
 	 * @see https://developer.wordpress.org/block-editor/reference-guides/block-api/block-registration/
 	 */
 	registerBlockType( '{{namespace}}/{{slug}}', {
-		/**
```
Comment (gziolo) [inline packages/create-block/lib/templates/es5/index.js.mustache:36]: The removed code is loading from `block.json` now.

**Review by ryanwelcher — APPROVED** [review]
LGTM! Really excited to get this out in the world!

**Comment by gziolo** [comment]
Included changes to the documentation with [c7be5ea](https://github.com/WordPress/gutenberg/pull/44185/commits/c7be5eaaccd81af15e055419bc115a9e2701f2f2) that reflect the most recent fix in WordPress core: https://core.trac.wordpress.org/changeset/54367.

**Comment by cbravobernal** [comment]
@ryanwelcher , could you please resolve the conflicts of this PR so we can land it? 
I guess you will have much more context than me on this task 😅 

Thank you!

**Comment by ryanwelcher** [comment]
@c4rl0sbr4v0  I've updated the PR. There is a change to a dependabot.yml file that I am not sure is supposed to be there though. Maybe I should revert that file?

**Comment by cbravobernal** [comment]
Thank you!
We can revert it, as the change is only indent and double comma changes. It seems to have been done by the linter. In case we don't revert it, it seems that there is a green check on the dependabot config file validation so should be ok too.

**Comment by gziolo** [comment]
Thank you for bringing it up to date. It should be good to land in a few days so it gets published to npm on Nov 2nd - one day after WP 6.1 is out. It feels like a perfect timing with Gutenberg releases 😃

**Pull request merged**